### PR TITLE
Fixing calendar events

### PR DESCRIPTION
**Pull request by TheAmericanMomo**
The calendar events had a bug with event deletion so I fixed this issue. Also took this time to implement CalendarEvents in a better way and made the UI more appealing!!!

Now we shouldnt have any issue when one of us deletes an event directly in firebase but doesn't remove pointers for the profiles that were registered to that event.

### CODE REVIEW

File: app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/filter/CalendarEventFilter.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+public class CalendarEventFilter implements EventFilter{
+    String profileId;
+
+    public CalendarEventFilter(String profileId){
+        this.profileId = profileId;
+    }
+
+    @Override
+    public boolean test(Event event) {
+        return event.getParticipants().contains(profileId);
```
Comment (DavidKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/filter/CalendarEventFilter.java:15]: Maybe I'm wrong but here should not we as well check if the profileId is or the organiser?

File: app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/list/EventsCalendarAdapter.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        // Format the date before showing it
+        registeredEventTime.setText(DateFormat.format("dd-MM-yyyy (HH:mm:ss)",
+                event.getDate()));
+
+
+        registeredEventTime.setGravity(Gravity.CENTER);
+        eventTitle.setGravity(Gravity.CENTER);
+
+        view.setOnClickListener(v -> EventActivity.openEvent(getContext(), event.getId()));
+        return view;
```
Comment (DavidKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/list/EventsCalendarAdapter.java:73]: Nice adapter, like the solution to use a custom adapter for this calendar view.
This is the way to go, especially if in the future we want to make other types of lists that look differently

File: app/src/main/java/com/sdpteam/connectout/profile/Profile.java [hunk 3]
```diff
@@ -72,32 +72,4 @@ public String getNameLowercase() {
     public enum Gender {
         FEMALE, MALE, OTHER
     }
-
-    public static class CalendarEvent {
```
Comment (DavidKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/profile/Profile.java:76]: Finally this whole mess is gone 🤩

**Review by DavidKalajdzic — COMMENTED** [review]
Thanks for this refactor of calendar, but as you can see by doing it you decreased the coverage rate, is it possible to check what is not tested anymore so we can stonk the leaderboard https://dslab.epfl.ch/teaching/sweng/proj ?

File: app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/list/EventsCalendarAdapter.java (hunk 2 above)
Comment (TheAmericanMomo) [reply app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/list/EventsCalendarAdapter.java:73]: Thanks boyiiiii!!!!!!

File: app/src/main/java/com/sdpteam/connectout/profile/Profile.java (hunk 3 above)
Comment (TheAmericanMomo) [reply app/src/main/java/com/sdpteam/connectout/profile/Profile.java:76]: Yes indeed, it came from a good intention, but there was obviously a better way of implementing all of this.

File: app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/filter/CalendarEventFilter.java (hunk 1 above)
Comment (TheAmericanMomo) [reply app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/filter/CalendarEventFilter.java:15]: By default, organizers are part of an event so this should be good

**Review by DavidKalajdzic — APPROVED** [review]

**Review by DavidKalajdzic — APPROVED** [review]

File: app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/filter/CalendarEventFilter.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package com.sdpteam.connectout.event.nearbyEvents.filter;
+
+import com.sdpteam.connectout.event.Event;
+import com.sdpteam.connectout.profile.Profile;
+
+public class CalendarEventFilter implements EventFilter{
```
Comment (DavidKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/filter/CalendarEventFilter.java:6]: This is not used by anyone, can delete this.

File: app/src/main/java/com/sdpteam/connectout/event/EventFirebaseDataSource.java [hunk 5]
```diff
@@ -83,6 +85,7 @@ public void onComplete(DatabaseError databaseError, boolean committed, DataSnaps
      * @return (CompletableFuture < Boolean >): completes to true if participant has joined the event.
      */
     public CompletableFuture<Boolean> joinEvent(String eventId, String participantId) {
+        new ProfileFirebaseDataSource().registerToEvent(new RegisteredEvent(eventId), participantId);
```
Comment (DavidKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/event/EventFirebaseDataSource.java:88]: Good idea, so the code Mikael did is still working 👍

File: app/src/main/java/com/sdpteam/connectout/profile/RegisteredEventsDataSource.java [hunk 6]
```diff
@@ -1,27 +1,13 @@
...
-import com.firebase.ui.database.FirebaseListOptions;
-
-import java.util.function.BiConsumer;
-import java.util.function.Consumer;
-import java.util.function.Function;
 
 public interface RegisteredEventsDataSource {
 
     /**
      * stores a new Profile.CalendarEvent (eventId, eventTitle and eventDate)
```
Comment (DavidKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/profile/RegisteredEventsDataSource.java:7]: I am allergic to this legacy code about Calendar.
Can you delete any mention of it in our code (do a Ctrl+Shift+F to search) so it does not confuse our team in the future.

**Review by DavidKalajdzic — CHANGES_REQUESTED** [review]
Good job handling the changes from master branch

File: app/src/main/java/com/sdpteam/connectout/event/EventFirebaseDataSource.java (hunk 5 above)
Comment (TheAmericanMomo) [reply app/src/main/java/com/sdpteam/connectout/event/EventFirebaseDataSource.java:88]: thanks, didn't have time to refactor everything

File: app/src/main/java/com/sdpteam/connectout/profile/RegisteredEventsDataSource.java (hunk 6 above)
Comment (TheAmericanMomo) [reply app/src/main/java/com/sdpteam/connectout/profile/RegisteredEventsDataSource.java:7]: does it give you a rash? :)
I'm on it

**Review by DavidKalajdzic — APPROVED** [review]

**Pull request merged**